internal/test/framework/framework.go

```diff
@@ -294,6 +300,62 @@ func (f *TestFramework) GetNode(externalIP string) (*v1.Node, error) {
 	return matchedNode, nil
 }
 
+// GetNode uses external IP and finds out the name associated with the node
+func (f *TestFramework) GetNodeName(externalIP string) (string, error) {
+	node, err := f.GetNode(externalIP)
+	if err != nil {
+		return "", fmt.Errorf("error while getting required kubernetes node object: %v", err)
+	}
+	return node.Name, nil
+}
+
+// RetrieveArtifacts should retrieve artifacts related the test run. Ideally this should retrieve all the logs related
+// to the Windows VM. This shouldn't return an error but should print the failures as log collection is nice to have
+// rather than a must have.
+// TODO: Think about how we can retrieve stdout from ansible out within this function
+func (f *TestFramework) RetrieveArtifacts() {
+	for i, vm := range f.WinVMs {
+		if vm == nil {
+			continue
+		}
+		if vm.GetCredentials() == nil {
+			log.Printf("error while getting credentials of the %d vm", i)
+			continue
+		}
+
+		instanceID := vm.GetCredentials().GetInstanceId()
+		if len(instanceID) == 0 {
+			log.Printf("error while getting instance id of the %d vm", i)
+			continue
+		}
+
+		externalIP := vm.GetCredentials().GetIPAddress()
+		if len(externalIP) == 0 {
+			log.Printf("error while getting external ip address of the vm with instance ID %s", instanceID)
+			continue
+		}
+
+		nodeName, err := f.GetNodeName(externalIP)
+		if err != nil {
+			log.Printf("error while getting node name associated with the vm %s: %v", instanceID, err)
+		}
+
+		// We want a format like "nodes/ip-10-0-141-99.ec2.internal/logs/wsu/kubelet"
+		localKubeletLogPath := filepath.Join(artifactDir, "nodes", nodeName, "logs", "wsu", "kubelet")
```

Comment:
What about the `wmcb` case? You cannot hard code this to `wsu`. More over there is going to be more than just the kubelet log here, so I suggest leaving out the `kubelet` directory.

Reply:
Ok, I'll remove it.